lib/rubocop/cop/chef/deprecation/chef_rewind.rb

```diff
@@ -61,47 +61,54 @@ class ChefRewind < Cop
             (send nil? ${:rewind :unwind} ... )
           PATTERN
 
+          extend AutoCorrector
           def on_send(node)
             rewind_gem_install?(node) do
-              add_offense(node, location: :expression, message: MSG, severity: :warning)
+              add_offense(node, message: MSG, severity: :warning) do |corrector|
+                fixer(node, corrector)
+              end
             end
 
             require_rewind?(node) do
-              add_offense(node, location: :expression, message: MSG, severity: :warning)
+              add_offense(node, message: MSG, severity: :warning) do |corrector|
+                fixer(node, corrector)
+              end
             end
 
             rewind_resources?(node) do
-              add_offense(node, location: :expression, message: MSG, severity: :warning)
+              add_offense(node, message: MSG, severity: :warning) do |corrector|
+                fixer(node, corrector)
+              end
             end
           end
 
           def on_block(node)
             match_property_in_resource?(:chef_gem, 'package_name', node) do |pkg_name|
-              add_offense(node, location: :expression, message: MSG, severity: :warning) if pkg_name.arguments&.first&.str_content == 'chef-rewind'
+              add_offense(node, message: MSG, severity: :warning) do |corrector|
+                fixer(node, corrector)
+              end if pkg_name.arguments&.first&.str_content == 'chef-rewind'
             end
           end
 
-          def autocorrect(node)
-            lambda do |corrector|
-              rewind_gem_install?(node) do
-                node = node.parent if node.parent&.block_type? # make sure we get the whole block not just the method in the block
-                corrector.remove(range_with_surrounding_space(range: node.loc.expression, side: :left))
-                return
-              end
+          def fixer(node, corrector)
+            rewind_gem_install?(node) do
```

Comment:
We should be able to extract this mess into each corrector not. It's one of the reasons they did the new autocorrect inline format.

Reply:
Fixed